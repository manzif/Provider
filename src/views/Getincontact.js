import React, { Component } from "react";
import PropTypes from 'prop-types'
import { toast } from "react-toastify";

import  Form  from "../components/common/form/Form"
import { connect } from 'react-redux';
import { addMessage } from '../redux/actions/messageActions';


export class Getincontact extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: {
        fullname: "",
        email: "",
        telephone: "",
        message: "",
      }
    }
  }

  componentWillReceiveProps(nexProps) {
    console.log(nexProps)
    if (nexProps.message.status === 200) {
      this.setState({ fullname: "", email: "", telephone: "", message: "" });
      return toast.success(nexProps.message.shoutout, {
        className: 'toast-success-container toast-success-container-after'})
    }
    if (nexProps.message.status === 400) {
      return toast.error(nexProps.message.message, {
        className: 'toast-error-container toast-error-container-after'})
    }
  }

 
  onChange = e => {
    this.setState({[e.target.name]: e.target.value});
  }
  onSubmit = e => {
    e.preventDefault();
    const message  = {
      fullname: this.state.fullname,
      email: this.state.email,
      telephone: this.state.telephone,
      message: this.state.message
    }
    this.props.addMessage(message);
  }

  render() {
      return (
        <>
        <div className="popup">
            <a href="/" className="popup__close">&times;</a>
                <div className="popup__content">
                    <Form className="submtest"
                    fullname={this.state.fullname}
                    email={this.state.email}
                    telephone={this.state.telephone}
                    message={this.state.message}
                    onChange={this.onChange}
                    onSubmit={this.onSubmit}
                    ></Form>
                </div>
            </div>
        </>
      );
    }
  }


  Getincontact.propTypes = {
    addMessage: PropTypes.func.isRequired,
    message: PropTypes.object.isRequired
  }
  export const mapStateToProps = state => ({
    message: state.messages.message
  });

export default connect(mapStateToProps, { addMessage })(Getincontact);
