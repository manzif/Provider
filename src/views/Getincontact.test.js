import React from "react";
import { shallow } from "enzyme";
import { Getincontact } from "./Getincontact";

describe("landing page", () => {
  const component = shallow(<Getincontact  />);
  it("renders without errors", () => {
    expect(component).toMatchSnapshot();
  });
});
