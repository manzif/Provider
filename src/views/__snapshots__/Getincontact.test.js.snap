// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components renders without errors 1`] = `ShallowWrapper {}`;
